imap_processing/swapi/tests/test_swapi_l1.py

```diff
@@ -1 +1,344 @@
+    for packet_file in packet_files:
+        data_list.extend(decom_packets(packet_file, packet_definition))
+    return data_list
+
+
+def test_check_for_bad_data():
+    """Test for bad data"""
+    # create test data for this test
+    total_sweeps = 3
+    ds = xr.Dataset(
+        {
+            "PLAN_ID_SCIENCE": xr.DataArray(np.full((total_sweeps * 12), 1)),
+            "SWEEP_TABLE": xr.DataArray(np.repeat(np.arange(total_sweeps), 12)),
+            "MODE": xr.DataArray(np.full((total_sweeps * 12), 2)),
+        }
+    )
+
+    # Check for no bad data
+    bad_data_indices = check_for_bad_data(ds)
+    assert len(bad_data_indices) == 0
+
+    # Check for bad MODE data.
+    # This test returns this indices because MODE has 0, 1 values
+    # for the first two sweeps.
+    # TODO: update test when we update MODE from HVENG to HVSCI
+    ds["MODE"] = xr.DataArray(np.repeat(np.arange(total_sweeps), 12))
+    bad_data_indices = check_for_bad_data(ds)
+    np.testing.assert_array_equal(bad_data_indices, np.arange(0, 24))
+
+    # Check for bad SWEEP_TABLE data.
+    # Reset MODE data and create first sweep to be mixed value
+    ds["MODE"] = xr.DataArray(np.full((total_sweeps * 12), 2))
+    ds["SWEEP_TABLE"][:12] = np.arange(0, 12)
+    np.testing.assert_array_equal(check_for_bad_data(ds), np.arange(0, 12))
+
+    ds["SWEEP_TABLE"][24:] = np.arange(0, 12)
+    expected = [
+        0,
+        1,
+        2,
+        3,
+        4,
+        5,
+        6,
+        7,
+        8,
+        9,
+        10,
+        11,
+        24,
+        25,
+        26,
+        27,
+        28,
+        29,
+        30,
+        31,
+        32,
+        33,
+        34,
+        35,
+    ]
+    np.testing.assert_array_equal(check_for_bad_data(ds), expected)
+
+    # Check for bad PLAN_ID_SCIENCE data.
+    ds["SWEEP_TABLE"] = xr.DataArray(np.repeat(np.arange(total_sweeps), 12))
+    ds["PLAN_ID_SCIENCE"][24 : total_sweeps * 12] = np.arange(0, 12)
+    np.testing.assert_array_equal(check_for_bad_data(ds), np.arange(24, 36))
+
+
+def test_decompress_count():
+    """Test for decompress count"""
+    # Test compressed and not overflow case
+    raw_value = np.array([[12]])
+    compression_flag = np.array([[1]])
+    expected_value = raw_value[0][0] * 16
+    returned_value = decompress_count(raw_value, compression_flag)
+    assert returned_value[0][0] == expected_value
+
+    # Test compressed and overflow case
+    raw_value = np.array([[0xFFFF]])
+    compression_flag = np.array([[1]])
+    expected_value = -1
+    returned_value = decompress_count(raw_value, compression_flag)
+    assert returned_value[0][0] == expected_value
+
+    # Test not compressed case
+    raw_value = np.array([[12]])
+    compression_flag = np.array([[0]])
+    expected_value = 12
+    returned_value = decompress_count(raw_value, compression_flag)
+    assert returned_value[0][0] == expected_value
+
+    # compressed + no-overflow, compressed + overflow, no compression
```

Comment:
Just an FYI that I think you can remove all of the above 3-blocks since they are really duplicates of this test that combines them all, and this is more robust in a sense.